### DISCUSSION TITLE
How to configure LSP to avoid "Format Request Failed"

### DISCUSSION THREAD

**Discussion by cnshsliu** [LunarVim/LunarVim#3875] [Q&A] [answered]
Hello freinds:

I have marksman installed and configured as:
```
local util = require 'lspconfig.util'
require("lvim.lsp.manager").setup("pyright", opts)
require("lvim.lsp.manager").setup("marksman", {
  cmd = { "marksman", "server" },
  filetypes = { "markdown", "telekasten" },
  root_dir = function(fname)
    local root_files = { '.marksman.toml' }
    return util.root_pattern(unpack(root_files))(fname) or util.find_git_ancestor(fname) or util.path.dirname(fname)
  end,
  single_file_support = true
})
```
edit a simple markdown file "a.md", and Mason info looks like
<img width="1192" alt="image" src="https://user-images.githubusercontent.com/2124836/220080877-0394c36d-5094-4e62-ac37-dd77c286d357.png">
we could see "marksman" is successfully installed.

and LSP info looks like:
<img width="841" alt="image" src="https://user-images.githubusercontent.com/2124836/220081065-5802a625-b768-4026-88f1-4498fcb8086d.png">
we could see marksman client is successfully attached.

and `ps -ef|grep marksman` looks like:
<img width="655" alt="image" src="https://user-images.githubusercontent.com/2124836/220081296-62c3933c-7b1d-4678-8061-575fb474589b.png">
we could see "marksman server" is running.

Everything seems okay, however, when I save the markdown file, 
Notify message shows that `[LSP] Format request failed, no matching language servers`

I already deal with this problem for hours, hope some one helps me out, thanks in advance!

**Comment by LostInTheLogs** [comment, marked as answer]
Does that server support formatting? See :LvimInfo

**Reply by cnshsliu** [reply]
:LvimInfo shows  
![image](https://user-images.githubusercontent.com/2124836/220354860-36daddcf-6d47-4d2d-bea6-386512b88669.png)

There are markdownlint, mdformat etc. are supported, but none is in Active: list.

Please let me know what could I do from here. I use nvim a lot and love lvim, but haveing very limited knowledge about Lua bring me a lot of trouble to understand how to configure it .  Thanks in advance.

**Reply by LostInTheLogs** [reply]
Send a full screenshot containing capabilities of the lsp server

**Reply by LostInTheLogs** [reply]
If formatting is not in the list, get a formatter https://www.lunarvim.org/docs/master/configuration/language-features/linting-and-formatting

**Reply by cnshsliu** [reply]
Thank you very much @LostNeophyte,  according to your direction, I checked the config.lua again and finally make it work by adding `  { command = "markdownlint", filetypes = { "markdown" } },`

```
local formatters = require "lvim.lsp.null-ls.formatters"
formatters.setup {
  { command = "black",        filetypes = { "python" } },
  { command = "isort",        filetypes = { "python" } },
  { command = "markdownlint", filetypes = { "markdown" } },
  {
    -- each formatter accepts a list of options identical to https://github.com/jose-elias-alvarez/null-ls.nvim/blob/main/doc/BUILTINS.md#Configuration
    command = "prettier",
    ---@usage arguments to pass to the formatter
    -- these cannot contain whitespaces, options such as `--line-width 80` become either `{'--line-width', '80'}` or `{'--line-width=80'}`
    extra_args = { "--print-with", "100" },
    ---@usage specify which filetypes to enable. By default a providers will attach to all the filetypes it supports.
    filetypes = { "typescript", "typescriptreact" },
  },
}

```
